src/components/TemplateLibrary/index.js

```diff
@@ -200,24 +206,33 @@ const LibraryComponent = () => {
           const customXmlPart = result.value[0];
           customXmlPart.getNodesAsync('*/*', result => {
             if (result.status === Office.AsyncResultStatus.Succeeded) {
+              let identifierExists = false;
               let newXml = XML_HEADER + `<templates xmlns="${CUSTOM_XML_NAMESPACE}">`;
               if (result.value.length > 0) {
                 for (let node=0; node < result.value.length; ++node) {
                   if (result.value[node].namespaceUri !== templateIdentifier) {
                     newXml += `<template xmlns="${result.value[node].namespaceUri}" />`;
                   }
+                  if(result.value[node].namespaceUri === templateIdentifier){
+                    identifierExists = true;
+                  }
                 }
               }
-              newXml += `<template xmlns="${templateIdentifier}" />`;
-              newXml += '</templates>';
-              Office.context.document.customXmlParts.getByNamespaceAsync(CUSTOM_XML_NAMESPACE, res => {
-                if (res.status === Office.AsyncResultStatus.Succeeded) {
-                  for (let index=0; index<res.value.length; ++index) {
-                    res.value[index].deleteAsync();
+              if(!identifierExists){
```

Comment:
In a separate PR, add rules for [block-spacing](https://eslint.org/docs/rules/space-before-blocks) so that this translates to `if (!identifierExists) {` (notice the spaces).

Reply:
@algomaster99 and/or @K-Kumar-01, May I work on this?

Reply:
Go for it!

Reply:
Sure thanks!